<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 1 How to install IOBR | IOBR (Immuno-Oncology Biological Research)</title>
  <meta name="description" content="The tutorial of IOBR package" />
  <meta name="generator" content="bookdown 0.35 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 1 How to install IOBR | IOBR (Immuno-Oncology Biological Research)" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="https://iobr.github.io/book/./fig/cover.png" />
  <meta property="og:description" content="The tutorial of IOBR package" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 1 How to install IOBR | IOBR (Immuno-Oncology Biological Research)" />
  
  <meta name="twitter:description" content="The tutorial of IOBR package" />
  <meta name="twitter:image" content="https://iobr.github.io/book/./fig/cover.png" />

<meta name="author" content="Dongqiang Zeng" />


<meta name="date" content="2023-09-08" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="rna-data-preprocessing.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">The tutorial of IOBR</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><strong>Introduction</strong></a>
<ul>
<li class="chapter" data-level="0.1" data-path="index.html"><a href="index.html#introduction-1"><i class="fa fa-check"></i><b>0.1</b> üìö Introduction</a></li>
<li class="chapter" data-level="0.2" data-path="index.html"><a href="index.html#license"><i class="fa fa-check"></i><b>0.2</b> üíø License</a></li>
<li class="chapter" data-level="0.3" data-path="index.html"><a href="index.html#publishment"><i class="fa fa-check"></i><b>0.3</b> üèÜ Publishment</a></li>
<li class="chapter" data-level="0.4" data-path="index.html"><a href="index.html#major-updates"><i class="fa fa-check"></i><b>0.4</b> ‚è≥ Major Updates</a></li>
<li class="chapter" data-level="0.5" data-path="index.html"><a href="index.html#reporting-bugs"><i class="fa fa-check"></i><b>0.5</b> ‚úâÔ∏è Reporting bugs</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="how-to-install-iobr.html"><a href="how-to-install-iobr.html"><i class="fa fa-check"></i><b>1</b> <strong>How to install IOBR</strong></a>
<ul>
<li class="chapter" data-level="1.1" data-path="how-to-install-iobr.html"><a href="how-to-install-iobr.html#install-dependency-packages"><i class="fa fa-check"></i><b>1.1</b> ‚è≥ Install Dependency Packages</a></li>
<li class="chapter" data-level="1.2" data-path="how-to-install-iobr.html"><a href="how-to-install-iobr.html#install-iobr-package"><i class="fa fa-check"></i><b>1.2</b> üíª Install IOBR package</a></li>
<li class="chapter" data-level="1.3" data-path="how-to-install-iobr.html"><a href="how-to-install-iobr.html#how-to-update-iobr"><i class="fa fa-check"></i><b>1.3</b> üìå How to update IOBR</a></li>
<li class="chapter" data-level="1.4" data-path="how-to-install-iobr.html"><a href="how-to-install-iobr.html#the-main-pipeline-of-iobr"><i class="fa fa-check"></i><b>1.4</b> üîç The main pipeline of IOBR</a></li>
<li class="chapter" data-level="1.5" data-path="how-to-install-iobr.html"><a href="how-to-install-iobr.html#main-functions-of-iobr"><i class="fa fa-check"></i><b>1.5</b> üóÇ Main Functions of IOBR</a></li>
<li class="chapter" data-level="1.6" data-path="how-to-install-iobr.html"><a href="how-to-install-iobr.html#current-working-environment"><i class="fa fa-check"></i><b>1.6</b> üó∫ Current working environment</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html"><i class="fa fa-check"></i><b>2</b> <strong>RNA Data preprocessing</strong></a>
<ul>
<li class="chapter" data-level="2.1" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html#loading-packages"><i class="fa fa-check"></i><b>2.1</b> Loading packages</a></li>
<li class="chapter" data-level="2.2" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html#download-array-data-using-geoquery"><i class="fa fa-check"></i><b>2.2</b> Download array data using <code>GEOquery</code></a></li>
<li class="chapter" data-level="2.3" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html#gene-annotation"><i class="fa fa-check"></i><b>2.3</b> Gene Annotation</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html#for-array-data-hgu133plus-2-affaymetrix"><i class="fa fa-check"></i><b>2.3.1</b> For Array data: HGU133PLUS-2 (Affaymetrix)</a></li>
<li class="chapter" data-level="2.3.2" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html#for-rnaseq-data"><i class="fa fa-check"></i><b>2.3.2</b> For RNAseq data</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html#identifying-outlier-samples"><i class="fa fa-check"></i><b>2.4</b> Identifying outlier samples</a></li>
<li class="chapter" data-level="2.5" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html#pca-analysis-of-molecular-subtypes"><i class="fa fa-check"></i><b>2.5</b> PCA analysis of molecular subtypes</a></li>
<li class="chapter" data-level="2.6" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html#batch-effect-correction"><i class="fa fa-check"></i><b>2.6</b> Batch effect correction</a></li>
<li class="chapter" data-level="2.7" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html#references"><i class="fa fa-check"></i><b>2.7</b> References</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="tumor-ecosystem-analysis.html"><a href="tumor-ecosystem-analysis.html"><i class="fa fa-check"></i><b>3</b> <strong>Tumor ecosystem analysis</strong></a>
<ul>
<li class="chapter" data-level="3.1" data-path="tumor-ecosystem-analysis.html"><a href="tumor-ecosystem-analysis.html#loading-packages-1"><i class="fa fa-check"></i><b>3.1</b> Loading packages</a></li>
<li class="chapter" data-level="3.2" data-path="tumor-ecosystem-analysis.html"><a href="tumor-ecosystem-analysis.html#downloading-data-for-example"><i class="fa fa-check"></i><b>3.2</b> Downloading data for example</a></li>
<li class="chapter" data-level="3.3" data-path="tumor-ecosystem-analysis.html"><a href="tumor-ecosystem-analysis.html#gene-annotation-hgu133plus-2-affaymetrix"><i class="fa fa-check"></i><b>3.3</b> Gene Annotation: HGU133PLUS-2 (Affaymetrix)</a></li>
<li class="chapter" data-level="3.4" data-path="tumor-ecosystem-analysis.html"><a href="tumor-ecosystem-analysis.html#determine-tme-subtype-of-gastric-cancer-using-tmeclassifier-r-package"><i class="fa fa-check"></i><b>3.4</b> Determine TME subtype of gastric cancer using TMEclassifier R package</a></li>
<li class="chapter" data-level="3.5" data-path="tumor-ecosystem-analysis.html"><a href="tumor-ecosystem-analysis.html#deg-analysis-method1"><i class="fa fa-check"></i><b>3.5</b> DEG analysis: method1</a></li>
<li class="chapter" data-level="3.6" data-path="tumor-ecosystem-analysis.html"><a href="tumor-ecosystem-analysis.html#gsea-analysis-based-on-differential-express-gene-analysis-results"><i class="fa fa-check"></i><b>3.6</b> GSEA analysis based on differential express gene analysis results</a></li>
<li class="chapter" data-level="3.7" data-path="tumor-ecosystem-analysis.html"><a href="tumor-ecosystem-analysis.html#deg-analysis-method2"><i class="fa fa-check"></i><b>3.7</b> DEG analysis: method2</a></li>
<li class="chapter" data-level="3.8" data-path="tumor-ecosystem-analysis.html"><a href="tumor-ecosystem-analysis.html#identifying-signatures-associated-with-tme-clusters"><i class="fa fa-check"></i><b>3.8</b> Identifying signatures associated with TME clusters</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="signature-and-relevant-phenotypes.html"><a href="signature-and-relevant-phenotypes.html"><i class="fa fa-check"></i><b>4</b> <strong>Signature and relevant phenotypes</strong></a>
<ul>
<li class="chapter" data-level="4.1" data-path="signature-and-relevant-phenotypes.html"><a href="signature-and-relevant-phenotypes.html#loading-packages-2"><i class="fa fa-check"></i><b>4.1</b> Loading packages</a></li>
<li class="chapter" data-level="4.2" data-path="signature-and-relevant-phenotypes.html"><a href="signature-and-relevant-phenotypes.html#downloading-data-for-example-1"><i class="fa fa-check"></i><b>4.2</b> Downloading data for example</a></li>
<li class="chapter" data-level="4.3" data-path="signature-and-relevant-phenotypes.html"><a href="signature-and-relevant-phenotypes.html#signature-score-estimation"><i class="fa fa-check"></i><b>4.3</b> Signature score estimation</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="signature-and-relevant-phenotypes.html"><a href="signature-and-relevant-phenotypes.html#signature-collection-of-iobr"><i class="fa fa-check"></i><b>4.3.1</b> Signature collection of IOBR</a></li>
<li class="chapter" data-level="4.3.2" data-path="signature-and-relevant-phenotypes.html"><a href="signature-and-relevant-phenotypes.html#estimated-by-pca-method"><i class="fa fa-check"></i><b>4.3.2</b> Estimated by PCA method</a></li>
<li class="chapter" data-level="4.3.3" data-path="signature-and-relevant-phenotypes.html"><a href="signature-and-relevant-phenotypes.html#estimated-by-ssgsea-methodology"><i class="fa fa-check"></i><b>4.3.3</b> Estimated by ssGSEA methodology</a></li>
<li class="chapter" data-level="4.3.4" data-path="signature-and-relevant-phenotypes.html"><a href="signature-and-relevant-phenotypes.html#estimated-by-zscore-function"><i class="fa fa-check"></i><b>4.3.4</b> Estimated by zscore function</a></li>
<li class="chapter" data-level="4.3.5" data-path="signature-and-relevant-phenotypes.html"><a href="signature-and-relevant-phenotypes.html#reference"><i class="fa fa-check"></i><b>4.3.5</b> Reference</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="signature-and-relevant-phenotypes.html"><a href="signature-and-relevant-phenotypes.html#identifying-features-associated-with-survival"><i class="fa fa-check"></i><b>4.4</b> Identifying features associated with survival</a></li>
<li class="chapter" data-level="4.5" data-path="signature-and-relevant-phenotypes.html"><a href="signature-and-relevant-phenotypes.html#visulization-using-heatmap"><i class="fa fa-check"></i><b>4.5</b> Visulization using heatmap</a></li>
<li class="chapter" data-level="4.6" data-path="signature-and-relevant-phenotypes.html"><a href="signature-and-relevant-phenotypes.html#focus-on-target-signatures"><i class="fa fa-check"></i><b>4.6</b> Focus on target signatures</a></li>
<li class="chapter" data-level="4.7" data-path="signature-and-relevant-phenotypes.html"><a href="signature-and-relevant-phenotypes.html#survival-analysis"><i class="fa fa-check"></i><b>4.7</b> Survival analysis</a></li>
<li class="chapter" data-level="4.8" data-path="signature-and-relevant-phenotypes.html"><a href="signature-and-relevant-phenotypes.html#batch-correlation-analysis"><i class="fa fa-check"></i><b>4.8</b> Batch correlation analysis</a></li>
<li class="chapter" data-level="4.9" data-path="signature-and-relevant-phenotypes.html"><a href="signature-and-relevant-phenotypes.html#visulization-of-correlations"><i class="fa fa-check"></i><b>4.9</b> Visulization of correlations</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html"><i class="fa fa-check"></i><b>5</b> <strong>TME deconvolution</strong></a>
<ul>
<li class="chapter" data-level="5.1" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#loading-packages-3"><i class="fa fa-check"></i><b>5.1</b> Loading packages</a></li>
<li class="chapter" data-level="5.2" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#downloading-data-for-example-2"><i class="fa fa-check"></i><b>5.2</b> Downloading data for example</a></li>
<li class="chapter" data-level="5.3" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#available-methods-to-decode-tme-contexture"><i class="fa fa-check"></i><b>5.3</b> Available Methods to Decode TME Contexture</a></li>
<li class="chapter" data-level="5.4" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#method-1-cibersort"><i class="fa fa-check"></i><b>5.4</b> Method 1: CIBERSORT</a></li>
<li class="chapter" data-level="5.5" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#method-2-epic"><i class="fa fa-check"></i><b>5.5</b> Method 2: EPIC</a></li>
<li class="chapter" data-level="5.6" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#method-3-mcpcounter"><i class="fa fa-check"></i><b>5.6</b> Method 3: MCPcounter</a></li>
<li class="chapter" data-level="5.7" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#method-4-xcell"><i class="fa fa-check"></i><b>5.7</b> Method 4: xCELL</a></li>
<li class="chapter" data-level="5.8" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#method-5-estimate"><i class="fa fa-check"></i><b>5.8</b> Method 5: ESTIMATE</a></li>
<li class="chapter" data-level="5.9" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#method-6-timer"><i class="fa fa-check"></i><b>5.9</b> Method 6: TIMER</a></li>
<li class="chapter" data-level="5.10" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#method-7-quantiseq"><i class="fa fa-check"></i><b>5.10</b> Method 7: quanTIseq</a></li>
<li class="chapter" data-level="5.11" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#method-8-ips"><i class="fa fa-check"></i><b>5.11</b> Method 8: IPS</a></li>
<li class="chapter" data-level="5.12" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#combination-of-above-deconvolution-results"><i class="fa fa-check"></i><b>5.12</b> Combination of above deconvolution results</a></a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="references-1.html"><a href="references-1.html"><i class="fa fa-check"></i><b>6</b> <strong>References</strong></a>
<ul>
<li class="chapter" data-level="6.1" data-path="references-1.html"><a href="references-1.html#tme-deconvolution-1"><i class="fa fa-check"></i><b>6.1</b> TME deconvolution</a></li>
<li class="chapter" data-level="6.2" data-path="references-1.html"><a href="references-1.html#tme-signatures"><i class="fa fa-check"></i><b>6.2</b> TME Signatures</a></li>
<li class="chapter" data-level="6.3" data-path="references-1.html"><a href="references-1.html#data-sets"><i class="fa fa-check"></i><b>6.3</b> Data sets</a></li>
<li class="chapter" data-level="6.4" data-path="references-1.html"><a href="references-1.html#others"><i class="fa fa-check"></i><b>6.4</b> Others</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/DongqiangZeng0808" target="blank">Welcome to my GitHub</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">IOBR (Immuno-Oncology Biological Research)</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="how-to-install-iobr" class="section level1 hasAnchor" number="1">
<h1><span class="header-section-number">Chapter 1</span> <strong>How to install IOBR</strong><a href="how-to-install-iobr.html#how-to-install-iobr" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="install-dependency-packages" class="section level2 hasAnchor" number="1.1">
<h2><span class="header-section-number">1.1</span> ‚è≥ Install Dependency Packages<a href="how-to-install-iobr.html#install-dependency-packages" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>It is essential that you have R 3.6.3 or above already installed on your computer or server. IOBR is a pipeline that utilizes many other R packages that are currently available from CRAN, Bioconductor and GitHub.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="how-to-install-iobr.html#cb1-1" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;BiocManager&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) <span class="fu">install.packages</span>(<span class="st">&quot;BiocManager&quot;</span>)</span>
<span id="cb1-2"><a href="how-to-install-iobr.html#cb1-2" tabindex="-1"></a>depens<span class="ot">&lt;-</span><span class="fu">c</span>(<span class="st">&#39;tibble&#39;</span>, <span class="st">&#39;survival&#39;</span>, <span class="st">&#39;survminer&#39;</span>, <span class="st">&#39;limma&#39;</span>, <span class="st">&quot;DESeq2&quot;</span>,<span class="st">&quot;devtools&quot;</span>, <span class="st">&#39;limSolve&#39;</span>, <span class="st">&#39;GSVA&#39;</span>, <span class="st">&#39;e1071&#39;</span>, <span class="st">&#39;preprocessCore&#39;</span>, </span>
<span id="cb1-3"><a href="how-to-install-iobr.html#cb1-3" tabindex="-1"></a>          <span class="st">&quot;devtools&quot;</span>, <span class="st">&quot;tidyHeatmap&quot;</span>, <span class="st">&quot;caret&quot;</span>, <span class="st">&quot;glmnet&quot;</span>, <span class="st">&quot;ppcor&quot;</span>,  <span class="st">&quot;timeROC&quot;</span>, <span class="st">&quot;pracma&quot;</span>, <span class="st">&quot;factoextra&quot;</span>, </span>
<span id="cb1-4"><a href="how-to-install-iobr.html#cb1-4" tabindex="-1"></a>          <span class="st">&quot;FactoMineR&quot;</span>, <span class="st">&quot;WGCNA&quot;</span>, <span class="st">&quot;patchwork&quot;</span>, <span class="st">&#39;ggplot2&#39;</span>, <span class="st">&quot;biomaRt&quot;</span>, <span class="st">&#39;ggpubr&#39;</span>)</span>
<span id="cb1-5"><a href="how-to-install-iobr.html#cb1-5" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(depens)){</span>
<span id="cb1-6"><a href="how-to-install-iobr.html#cb1-6" tabindex="-1"></a>  depen<span class="ot">&lt;-</span>depens[i]</span>
<span id="cb1-7"><a href="how-to-install-iobr.html#cb1-7" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(depen, <span class="at">quietly =</span> <span class="cn">TRUE</span>))  BiocManager<span class="sc">::</span><span class="fu">install</span>(depen,<span class="at">update =</span> <span class="cn">FALSE</span>)</span>
<span id="cb1-8"><a href="how-to-install-iobr.html#cb1-8" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="install-iobr-package" class="section level2 hasAnchor" number="1.2">
<h2><span class="header-section-number">1.2</span> üíª Install IOBR package<a href="#install-iobr-package" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>When the dependent environments are built, users are able to install IOBR from github by typing the following code into your R session:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="how-to-install-iobr.html#cb2-1" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;IOBR&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))  devtools<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">&quot;IOBR/IOBR&quot;</span>)</span>
<span id="cb2-2"><a href="how-to-install-iobr.html#cb2-2" tabindex="-1"></a></span>
<span id="cb2-3"><a href="how-to-install-iobr.html#cb2-3" tabindex="-1"></a><span class="fu">library</span>(IOBR)</span></code></pre></div>
</div>
<div id="how-to-update-iobr" class="section level2 hasAnchor" number="1.3">
<h2><span class="header-section-number">1.3</span> üìå How to update IOBR<a href="#how-to-update-iobr" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="how-to-install-iobr.html#cb3-1" tabindex="-1"></a><span class="fu">detach</span>(<span class="st">&quot;package:IOBR&quot;</span>)</span>
<span id="cb3-2"><a href="how-to-install-iobr.html#cb3-2" tabindex="-1"></a>path<span class="ot">&lt;-</span><span class="fu">.libPaths</span>()</span>
<span id="cb3-3"><a href="how-to-install-iobr.html#cb3-3" tabindex="-1"></a><span class="fu">remove.packages</span>(<span class="fu">c</span>(<span class="st">&#39;IOBR&#39;</span>), <span class="at">lib=</span><span class="fu">file.path</span>(path))</span>
<span id="cb3-4"><a href="how-to-install-iobr.html#cb3-4" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">&quot;IOBR/IOBR&quot;</span>)</span></code></pre></div>
</div>
<div id="the-main-pipeline-of-iobr" class="section level2 hasAnchor" number="1.4">
<h2><span class="header-section-number">1.4</span> üîç The main pipeline of IOBR<a href="#the-main-pipeline-of-iobr" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:flowchart"></span>
<img src="fig/IOBR-Package.png" alt="The main pipeline of IOBR" width="95%" />
<p class="caption">
Figure 1.1: The main pipeline of IOBR
</p>
</div>
</div>
<div id="main-functions-of-iobr" class="section level2 hasAnchor" number="1.5">
<h2><span class="header-section-number">1.5</span> üóÇ Main Functions of IOBR<a href="#main-functions-of-iobr" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li><div style="color:green">
<strong>Data Preparation: data annotation and transformation</strong>
</div>
<ul>
<li><code>count2tpm()</code>: transform count data of RNA sequencing into TPM data.</li>
<li><code>anno_eset()</code>: annotate the normalized genes expression matrix, including RNAseq and array (Affymetrix or Illumina).</li>
<li><code>remove_duplicate_genes()</code>: remove the genes annotated with the duplicated symbol after normalization and retain only the symbol with highest expression level.</li>
<li><code>mouse2human_eset()</code>: Converting muouse gene symbol to human gene symbol of expression set.</li>
<li><code>find_outlier_samples()</code>: Waiting for updates‚Ä¶</li>
<li><code>remove_batcheffect()</code>: Waiting for updates‚Ä¶
</br></li>
</ul></li>
<li><div style="color:green">
<strong>TME Deconvolution Module: integrate multiple algorithms to decode immune contexture</strong>
</div>
<ul>
<li><code>deconvo_tme()</code>: decode the TME infiltration with different deconvolution methodologies, based on bulk RNAseq, microarray or single cell RNAseq data.</li>
<li><code>generateRef()</code>: generate a novel gene reference matrix for a specific feature such as infiltrating cell, through the SVR and lsei algorithm.
</br></li>
</ul></li>
<li><div style="color:green">
<strong>Signature Module: calculatint signature scores, estimate phenotype related signatures and corresponding genes, and evaluate signatures generated from single-cell RNA sequencing data </strong>
</div>
<ul>
<li><code>calculate_sig_score()</code>: estimate the interested signatures enrolled in IOBR R package, which involves TME-associated, tumor-metabolism, and tumor-intrinsic signatures.</li>
<li><code>feature_manipulation()</code>: manipulate features including the cell fraction and signatures generated from multi-omics data for latter analysis and model construction. Remove missing values, outliers and variables without significant variance.</li>
<li><code>format_signatures()</code>: generate the object for <code>calculate_sig_score()</code> function, by inputting a data frame with signatures as column names of corresponding gene sets, and return a list contain the signature information for calculating multiple signature scores.</li>
<li><code>format_msigdb()</code>: transform the signature gene sets data with gmt format, which is not included in the signature collection and might be downloaded in the MSgiDB website, into the object of <code>calculate_sig_score()</code>function.</li>
<li><code>sig_gsea()</code>: Waiting for updates‚Ä¶
</br></li>
</ul></li>
<li><div style="color:green">
<strong>Batch Analysis and Visualization: batch survival analysis and batch correlation analysis and other batch statistical analyses </strong>
</div>
<ul>
<li><code>batch_surv()</code>: batch survival analysis of multiple continuous variables including varied signature scores.</li>
<li><code>subgroup_survival()</code>: batch survival analysis of multiple categorized variables with different number of subgroups.</li>
<li><code>batch_cor()</code>: batch analysis of correlation between two continuous variables using Pearson correlation coefficient or Spearman‚Äôs rank correlation coefficient .</li>
<li><code>batch_wilcoxon()</code>: conduct batch wilcoxon analyses of binary variables.</li>
<li><code>batch_pcc()</code>: batch analyses of Partial Correlation coefficient(PCC) between continuous variables and minimize the interference derived from confounding factors.</li>
<li><code>iobr_cor_plot()</code>: visualization of batch correlation analysis of signatures from ‚Äòsig_group‚Äô. Visualize the correlation between signature or phenotype with expression of gene sets in target signature is also supported.</li>
<li><code>cell_bar_plot()</code>: batch visualization of TME cell fraction, supporting input of deconvolution results from ‚ÄòCIBERSORT‚Äô, ‚ÄòEPIC‚Äô and ‚ÄòquanTIseq‚Äô methodologies to further compare the TME cell distributions within one sample or among different samples.</li>
<li><code>iobr_pca()</code>: The iobr_pca function performs Principal Component Analysis (PCA), which reduces the dimensionality of data while maintaining most of the original variance, and visualizes the PCA results on a scatter plot.</li>
<li><code>iobr_cor_plot()</code>: Integrative correlation between phenotype and features.</li>
<li><code>iobr_deg()</code>: Waiting for updates‚Ä¶</li>
<li><code>get_cor()</code>: Waiting for updates‚Ä¶</li>
<li><code>roc_time()</code>: Waiting for updates‚Ä¶</li>
<li><code>sig_box()</code>: Waiting for updates‚Ä¶</li>
<li><code>sig_heatmap()</code>: Waiting for updates‚Ä¶</li>
<li><code>sig_forest()</code>: Waiting for updates‚Ä¶</li>
<li><code>sig_roc()</code>: Waiting for updates‚Ä¶</li>
<li><code>sig_surv_plot()</code>: Waiting for updates‚Ä¶</li>
<li><code>find_markers_in_bulk()</code>: Waiting for updates‚Ä¶</li>
</ul></li>
</ul>
<p></br></p>
<ul>
<li><div style="color:green">
<strong>Signature Associated Mutation Module: identify and analyze mutations relevant to targeted signatures</strong>
</div>
<ul>
<li><code>make_mut_matrix()</code>: transform the mutation data with MAF format(contain the columns of gene ID and the corresponding gene alterations which including SNP, indel and frameshift) into a mutation matrix in a suitable manner for further investigating signature relevant mutations.</li>
<li><code>find_mutations()</code>: identify mutations associated with a distinct phenotype or signature.
</br></li>
</ul></li>
<li><div style="color:green">
<strong>Model Construction Module: feature selection and fast model construct to predict clinical phenotype</strong>
</div>
<ul>
<li><code>BinomialModel()</code>: select features and construct a model to predict a binary phenotype.</li>
<li><code>PrognosticMode()</code>: select features and construct a model to predict clinical survival outcome.
</br></li>
</ul></li>
</ul>
</div>
<div id="current-working-environment" class="section level2 hasAnchor" number="1.6">
<h2><span class="header-section-number">1.6</span> üó∫ Current working environment<a href="#current-working-environment" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="how-to-install-iobr.html#cb4-1" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.2.0 (2022-04-22 ucrt)
## Platform: x86_64-w64-mingw32/x64 (64-bit)
## Running under: Windows 10 x64 (build 19045)
## 
## Matrix products: default
## 
## locale:
## [1] LC_COLLATE=Chinese (Simplified)_China.utf8 
## [2] LC_CTYPE=Chinese (Simplified)_China.utf8   
## [3] LC_MONETARY=Chinese (Simplified)_China.utf8
## [4] LC_NUMERIC=C                               
## [5] LC_TIME=Chinese (Simplified)_China.utf8    
## 
## attached base packages:
## [1] grid      stats     graphics  grDevices utils     datasets  methods  
## [8] base     
## 
## other attached packages:
##  [1] IOBR_0.99.9           survminer_0.4.9       patchwork_1.1.1      
##  [4] clusterProfiler_4.4.3 survival_3.3-1        ggpubr_0.4.0         
##  [7] ggplot2_3.4.2         dplyr_1.1.2           tibble_3.2.1         
## [10] tidyHeatmap_1.8.1     ComplexHeatmap_2.15.4
## 
## loaded via a namespace (and not attached):
##   [1] utf8_1.2.2                  tidyselect_1.2.0           
##   [3] RSQLite_2.2.14              AnnotationDbi_1.58.0       
##   [5] BiocParallel_1.30.3         lpSolve_5.6.15             
##   [7] scatterpie_0.1.7            ScaledMatrix_1.4.0         
##   [9] munsell_0.5.0               codetools_0.2-18           
##  [11] preprocessCore_1.58.0       withr_2.5.0                
##  [13] colorspace_2.0-3            GOSemSim_2.22.0            
##  [15] Biobase_2.56.0              limSolve_1.5.6             
##  [17] highr_0.9                   knitr_1.39                 
##  [19] rstudioapi_0.13             SingleCellExperiment_1.18.0
##  [21] stats4_4.2.0                ggsignif_0.6.3             
##  [23] DOSE_3.22.0                 MatrixGenerics_1.8.0       
##  [25] GenomeInfoDbData_1.2.8      KMsurv_0.1-5               
##  [27] polyclip_1.10-0             bit64_4.0.5                
##  [29] farver_2.1.0                rhdf5_2.40.0               
##  [31] downloader_0.4              treeio_1.21.2              
##  [33] vctrs_0.6.2                 generics_0.1.3             
##  [35] xfun_0.40                   R6_2.5.1                   
##  [37] doParallel_1.0.17           GenomeInfoDb_1.34.4        
##  [39] clue_0.3-61                 graphlayouts_0.8.0         
##  [41] rsvd_1.0.5                  locfit_1.5-9.5             
##  [43] rhdf5filters_1.8.0          gridGraphics_0.5-1         
##  [45] bitops_1.0-7                cachem_1.0.6               
##  [47] fgsea_1.22.0                DelayedArray_0.22.0        
##  [49] assertthat_0.2.1            scales_1.2.0               
##  [51] ggraph_2.0.5                enrichplot_1.16.1          
##  [53] googlesheets4_1.0.0         gtable_0.3.1               
##  [55] beachmat_2.12.0             tidygraph_1.2.1            
##  [57] rlang_1.1.0                 genefilter_1.78.0          
##  [59] GlobalOptions_0.1.2         splines_4.2.0              
##  [61] lazyeval_0.2.2              rstatix_0.7.0              
##  [63] gargle_1.2.0                broom_0.8.0                
##  [65] yaml_2.3.5                  reshape2_1.4.4             
##  [67] abind_1.4-5                 modelr_0.1.8               
##  [69] backports_1.4.1             qvalue_2.28.0              
##  [71] tools_4.2.0                 bookdown_0.35              
##  [73] ggplotify_0.1.0             ellipsis_0.3.2             
##  [75] jquerylib_0.1.4             RColorBrewer_1.1-3         
##  [77] proxy_0.4-27                BiocGenerics_0.42.0        
##  [79] Rcpp_1.0.9                  plyr_1.8.7                 
##  [81] sparseMatrixStats_1.8.0     zlibbioc_1.42.0            
##  [83] purrr_0.3.4                 RCurl_1.98-1.7             
##  [85] GetoptLong_1.0.5            viridis_0.6.2              
##  [87] cowplot_1.1.1               S4Vectors_0.34.0           
##  [89] zoo_1.8-10                  SummarizedExperiment_1.26.1
##  [91] haven_2.5.0                 ggrepel_0.9.1              
##  [93] cluster_2.1.3               fs_1.5.2                   
##  [95] magrittr_2.0.3              data.table_1.14.2          
##  [97] DO.db_2.9                   circlize_0.4.15            
##  [99] reprex_2.0.1                googledrive_2.0.0          
## [101] matrixStats_0.62.0          GSVA_1.44.2                
## [103] hms_1.1.1                   evaluate_0.15              
## [105] xtable_1.8-4                XML_3.99-0.10              
## [107] readxl_1.4.0                IRanges_2.30.0             
## [109] gridExtra_2.3               shape_1.4.6                
## [111] compiler_4.2.0              shadowtext_0.1.2           
## [113] crayon_1.5.2                htmltools_0.5.2            
## [115] ggfun_0.0.6                 tzdb_0.3.0                 
## [117] aplot_0.1.6                 tidyr_1.2.0                
## [119] geneplotter_1.74.0          lubridate_1.8.0            
## [121] DBI_1.1.2                   tweenr_1.0.2               
## [123] corrplot_0.92               dbplyr_2.2.0               
## [125] MASS_7.3-56                 Matrix_1.5-4.1             
## [127] car_3.1-0                   readr_2.1.2                
## [129] cli_3.4.1                   quadprog_1.5-8             
## [131] parallel_4.2.0              igraph_1.3.2               
## [133] GenomicRanges_1.48.0        forcats_0.5.1              
## [135] pkgconfig_2.0.3             km.ci_0.5-6                
## [137] xml2_1.3.3                  foreach_1.5.2              
## [139] ggtree_3.4.4                annotate_1.74.0            
## [141] bslib_0.3.1                 XVector_0.36.0             
## [143] rvest_1.0.2                 yulab.utils_0.0.4          
## [145] stringr_1.4.0               digest_0.6.29              
## [147] graph_1.74.0                Biostrings_2.64.0          
## [149] rmarkdown_2.14              cellranger_1.1.0           
## [151] fastmatch_1.1-3             tidytree_0.3.9             
## [153] survMisc_0.5.6              dendextend_1.15.2          
## [155] DelayedMatrixStats_1.18.0   GSEABase_1.58.0            
## [157] rjson_0.2.21                nlme_3.1-157               
## [159] lifecycle_1.0.3             jsonlite_1.8.0             
## [161] Rhdf5lib_1.18.2             carData_3.0-5              
## [163] viridisLite_0.4.1           limma_3.52.1               
## [165] fansi_1.0.3                 pillar_1.9.0               
## [167] lattice_0.20-45             KEGGREST_1.36.2            
## [169] fastmap_1.1.0               httr_1.4.3                 
## [171] GO.db_3.15.0                emo_0.0.0.9000             
## [173] glue_1.6.2                  png_0.1-7                  
## [175] iterators_1.0.14            glmnet_4.1-4               
## [177] bit_4.0.4                   HDF5Array_1.24.1           
## [179] ggforce_0.3.3               class_7.3-20               
## [181] stringi_1.7.6               sass_0.4.1                 
## [183] blob_1.2.3                  BiocSingular_1.12.0        
## [185] DESeq2_1.36.0               memoise_2.0.1              
## [187] irlba_2.3.5                 ape_5.6-2                  
## [189] tidyverse_1.3.2             e1071_1.7-11</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="rna-data-preprocessing.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/DongqiangZeng0808/install-iobr.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
