<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 4 Signature Score Calculation | IOBR (Immuno-Oncology Biological Research)</title>
  <meta name="description" content="The tutorial of IOBR package" />
  <meta name="generator" content="bookdown 0.36 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 4 Signature Score Calculation | IOBR (Immuno-Oncology Biological Research)" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="https://iobr.github.io/book/./fig/cover.png" />
  <meta property="og:description" content="The tutorial of IOBR package" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 4 Signature Score Calculation | IOBR (Immuno-Oncology Biological Research)" />
  
  <meta name="twitter:description" content="The tutorial of IOBR package" />
  <meta name="twitter:image" content="https://iobr.github.io/book/./fig/cover.png" />

<meta name="author" content="Dongqiang Zeng, Yiran Fang" />


<meta name="date" content="2023-12-21" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="rna-data-preprocessing.html"/>
<link rel="next" href="tme-deconvolution.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">The tutorial of IOBR</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><strong>Introduction</strong></a>
<ul>
<li class="chapter" data-level="0.1" data-path="index.html"><a href="index.html#introduction-1"><i class="fa fa-check"></i><b>0.1</b> üìö Introduction</a></li>
<li class="chapter" data-level="0.2" data-path="index.html"><a href="index.html#license"><i class="fa fa-check"></i><b>0.2</b> üíø License</a></li>
<li class="chapter" data-level="0.3" data-path="index.html"><a href="index.html#previous-publication"><i class="fa fa-check"></i><b>0.3</b> üèÜ Previous publication</a></li>
<li class="chapter" data-level="0.4" data-path="index.html"><a href="index.html#major-updates"><i class="fa fa-check"></i><b>0.4</b> ‚è≥ Major Updates</a></li>
<li class="chapter" data-level="0.5" data-path="index.html"><a href="index.html#reporting-bugs"><i class="fa fa-check"></i><b>0.5</b> ‚úâÔ∏è Reporting bugs</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="how-to-install-iobr.html"><a href="how-to-install-iobr.html"><i class="fa fa-check"></i><b>1</b> <strong>How to install IOBR</strong></a>
<ul>
<li class="chapter" data-level="1.1" data-path="how-to-install-iobr.html"><a href="how-to-install-iobr.html#install-dependency-packages"><i class="fa fa-check"></i><b>1.1</b> ‚è≥ Install Dependency Packages</a></li>
<li class="chapter" data-level="1.2" data-path="how-to-install-iobr.html"><a href="how-to-install-iobr.html#install-iobr-package"><i class="fa fa-check"></i><b>1.2</b> üíª Install IOBR package</a></li>
<li class="chapter" data-level="1.3" data-path="how-to-install-iobr.html"><a href="how-to-install-iobr.html#how-to-update-iobr"><i class="fa fa-check"></i><b>1.3</b> üìç How to update IOBR</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="how-to-use-iobr.html"><a href="how-to-use-iobr.html"><i class="fa fa-check"></i><b>2</b> <strong>How to use IOBR</strong></a>
<ul>
<li class="chapter" data-level="2.1" data-path="how-to-use-iobr.html"><a href="how-to-use-iobr.html#the-main-pipeline-of-iobr"><i class="fa fa-check"></i><b>2.1</b> üîç The main pipeline of IOBR</a></li>
<li class="chapter" data-level="2.2" data-path="how-to-use-iobr.html"><a href="how-to-use-iobr.html#main-functions-of-iobr"><i class="fa fa-check"></i><b>2.2</b> üóÇ Main Functions of IOBR</a></li>
<li class="chapter" data-level="2.3" data-path="how-to-use-iobr.html"><a href="how-to-use-iobr.html#current-working-environment"><i class="fa fa-check"></i><b>2.3</b> üåç Current working environment</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html"><i class="fa fa-check"></i><b>3</b> <strong>RNA Data preprocessing</strong></a>
<ul>
<li class="chapter" data-level="3.1" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html#loading-packages"><i class="fa fa-check"></i><b>3.1</b> Loading packages</a></li>
<li class="chapter" data-level="3.2" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html#download-array-data-using-geoquery"><i class="fa fa-check"></i><b>3.2</b> Download array data using <code>GEOquery</code></a></li>
<li class="chapter" data-level="3.3" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html#gene-annotation"><i class="fa fa-check"></i><b>3.3</b> Gene Annotation</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html#for-array-data-hgu133plus-2-affaymetrix"><i class="fa fa-check"></i><b>3.3.1</b> For Array data: HGU133PLUS-2 (Affaymetrix)</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html#download-rnaseq-data-using-ucscxenatools"><i class="fa fa-check"></i><b>3.4</b> Download RNAseq data using <code>UCSCXenaTools</code></a></li>
<li class="chapter" data-level="3.5" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html#normalization-and-gene-annotation"><i class="fa fa-check"></i><b>3.5</b> Normalization and Gene annotation</a></li>
<li class="chapter" data-level="3.6" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html#identifying-outlier-samples"><i class="fa fa-check"></i><b>3.6</b> Identifying outlier samples</a></li>
<li class="chapter" data-level="3.7" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html#pca-analysis-of-molecular-subtypes"><i class="fa fa-check"></i><b>3.7</b> PCA analysis of molecular subtypes</a></li>
<li class="chapter" data-level="3.8" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html#batch-effect-correction"><i class="fa fa-check"></i><b>3.8</b> Batch effect correction</a>
<ul>
<li class="chapter" data-level="3.8.1" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html#for-microarray-data"><i class="fa fa-check"></i><b>3.8.1</b> For microarray data</a></li>
<li class="chapter" data-level="3.8.2" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html#for-rnaseq-count-data"><i class="fa fa-check"></i><b>3.8.2</b> For RNAseq count data</a></li>
</ul></li>
<li class="chapter" data-level="3.9" data-path="rna-data-preprocessing.html"><a href="rna-data-preprocessing.html#references"><i class="fa fa-check"></i><b>3.9</b> References</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html"><i class="fa fa-check"></i><b>4</b> <strong>Signature Score Calculation</strong></a>
<ul>
<li class="chapter" data-level="4.1" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html#loading-packages-1"><i class="fa fa-check"></i><b>4.1</b> Loading packages</a></li>
<li class="chapter" data-level="4.2" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html#downloading-data-for-example"><i class="fa fa-check"></i><b>4.2</b> Downloading data for example</a></li>
<li class="chapter" data-level="4.3" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html#signature-score-estimation"><i class="fa fa-check"></i><b>4.3</b> Signature score estimation</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html#signature-collection-of-iobr"><i class="fa fa-check"></i><b>4.3.1</b> Signature collection of IOBR</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html#estimation-of-signature-using-pca-method"><i class="fa fa-check"></i><b>4.4</b> Estimation of signature using PCA method</a></li>
<li class="chapter" data-level="4.5" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html#estimated-using-the-ssgsea-methodology"><i class="fa fa-check"></i><b>4.5</b> Estimated using the ssGSEA methodology</a></li>
<li class="chapter" data-level="4.6" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html#calculated-using-the-z-score-function."><i class="fa fa-check"></i><b>4.6</b> Calculated using the z-score function.</a></li>
<li class="chapter" data-level="4.7" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html#calculated-using-all-three-methods-at-the-same-time"><i class="fa fa-check"></i><b>4.7</b> Calculated using all three methods at the same time</a></li>
<li class="chapter" data-level="4.8" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html#how-to-customise-the-signature-gene-list-for-calculate_signature_score"><i class="fa fa-check"></i><b>4.8</b> How to customise the signature gene list for <code>calculate_signature_score</code></a>
<ul>
<li class="chapter" data-level="4.8.1" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html#method-1-use-excel-for-storage-and-construction"><i class="fa fa-check"></i><b>4.8.1</b> Method-1: Use excel for storage and construction</a></li>
<li class="chapter" data-level="4.8.2" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html#method-2-build-the-list-structure-directly"><i class="fa fa-check"></i><b>4.8.2</b> Method-2: Build the list structure directly</a></li>
<li class="chapter" data-level="4.8.3" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html#method3-add-the-new-signature-to-the-existing-gene-list"><i class="fa fa-check"></i><b>4.8.3</b> Method3: Add the new signature to the existing gene list</a></li>
</ul></li>
<li class="chapter" data-level="4.9" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html#how-to-export-gene-signature"><i class="fa fa-check"></i><b>4.9</b> How to export gene signature</a></li>
<li class="chapter" data-level="4.10" data-path="signature-score-calculation.html"><a href="signature-score-calculation.html#references-1"><i class="fa fa-check"></i><b>4.10</b> References</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html"><i class="fa fa-check"></i><b>5</b> <strong>TME deconvolution</strong></a>
<ul>
<li class="chapter" data-level="5.1" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#loading-packages-2"><i class="fa fa-check"></i><b>5.1</b> Loading packages</a></li>
<li class="chapter" data-level="5.2" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#downloading-data-for-example-1"><i class="fa fa-check"></i><b>5.2</b> Downloading data for example</a></li>
<li class="chapter" data-level="5.3" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#available-methods-to-decode-tme-contexture"><i class="fa fa-check"></i><b>5.3</b> Available Methods to Decode TME Contexture</a></li>
<li class="chapter" data-level="5.4" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#method-1-cibersort"><i class="fa fa-check"></i><b>5.4</b> Method 1: CIBERSORT</a></li>
<li class="chapter" data-level="5.5" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#method-2-epic"><i class="fa fa-check"></i><b>5.5</b> Method 2: EPIC</a></li>
<li class="chapter" data-level="5.6" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#method-3-mcpcounter"><i class="fa fa-check"></i><b>5.6</b> Method 3: MCPcounter</a></li>
<li class="chapter" data-level="5.7" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#method-4-xcell"><i class="fa fa-check"></i><b>5.7</b> Method 4: xCELL</a></li>
<li class="chapter" data-level="5.8" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#method-5-estimate"><i class="fa fa-check"></i><b>5.8</b> Method 5: ESTIMATE</a></li>
<li class="chapter" data-level="5.9" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#method-6-timer"><i class="fa fa-check"></i><b>5.9</b> Method 6: TIMER</a></li>
<li class="chapter" data-level="5.10" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#method-7-quantiseq"><i class="fa fa-check"></i><b>5.10</b> Method 7: quanTIseq</a></li>
<li class="chapter" data-level="5.11" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#method-8-ips"><i class="fa fa-check"></i><b>5.11</b> Method 8: IPS</a></li>
<li class="chapter" data-level="5.12" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#combination-of-above-deconvolution-results"><i class="fa fa-check"></i><b>5.12</b> Combination of above deconvolution results</a></a></li>
<li class="chapter" data-level="5.13" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#how-to-customise-the-signature-matrix-for-svr-and-lesi-algorithm"><i class="fa fa-check"></i><b>5.13</b> How to customise the signature matrix for <code>SVR</code> and <code>lesi</code> algorithm</a></li>
<li class="chapter" data-level="5.14" data-path="tme-deconvolution.html"><a href="tme-deconvolution.html#references-2"><i class="fa fa-check"></i><b>5.14</b> References</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html"><i class="fa fa-check"></i><b>6</b> <strong>Signature Score and Relevant phenotypes</strong></a>
<ul>
<li class="chapter" data-level="6.1" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#loading-packages-3"><i class="fa fa-check"></i><b>6.1</b> Loading packages</a></li>
<li class="chapter" data-level="6.2" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#downloading-data-for-example-2"><i class="fa fa-check"></i><b>6.2</b> Downloading data for example</a></li>
<li class="chapter" data-level="6.3" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#gene-annotation-1"><i class="fa fa-check"></i><b>6.3</b> Gene Annotation</a></li>
<li class="chapter" data-level="6.4" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#estimation-of-signatures"><i class="fa fa-check"></i><b>6.4</b> Estimation of signatures</a></li>
<li class="chapter" data-level="6.5" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#combining-score-data-and-phenotype-data"><i class="fa fa-check"></i><b>6.5</b> Combining score data and phenotype data</a></li>
<li class="chapter" data-level="6.6" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#identifying-features-associated-with-survival"><i class="fa fa-check"></i><b>6.6</b> Identifying features associated with survival</a></li>
<li class="chapter" data-level="6.7" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#visulization-using-heatmap"><i class="fa fa-check"></i><b>6.7</b> Visulization using heatmap</a></li>
<li class="chapter" data-level="6.8" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#focus-on-target-signatures"><i class="fa fa-check"></i><b>6.8</b> Focus on target signatures</a></li>
<li class="chapter" data-level="6.9" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#survival-analysis-and-visulization"><i class="fa fa-check"></i><b>6.9</b> Survival analysis and visulization</a>
<ul>
<li class="chapter" data-level="6.9.1" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#kaplan-meier-plot"><i class="fa fa-check"></i><b>6.9.1</b> Kaplan-Meier plot</a></li>
<li class="chapter" data-level="6.9.2" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#time-dependent-roc-curve"><i class="fa fa-check"></i><b>6.9.2</b> Time-Dependent ROC curve</a></li>
</ul></li>
<li class="chapter" data-level="6.10" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#batch-correlation-analysis"><i class="fa fa-check"></i><b>6.10</b> Batch correlation analysis</a>
<ul>
<li class="chapter" data-level="6.10.1" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#finding-continuity-variables-associated-with-signatures"><i class="fa fa-check"></i><b>6.10.1</b> Finding continuity variables associated with signatures</a></li>
<li class="chapter" data-level="6.10.2" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#identifying-category-variables-linked-to-signatures"><i class="fa fa-check"></i><b>6.10.2</b> Identifying Category Variables Linked to Signatures</a></li>
<li class="chapter" data-level="6.10.3" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#for-multicategorical-variables-2-subgroups"><i class="fa fa-check"></i><b>6.10.3</b> For multicategorical variables (&gt;2 subgroups)</a></li>
</ul></li>
<li class="chapter" data-level="6.11" data-path="signature-score-and-relevant-phenotypes.html"><a href="signature-score-and-relevant-phenotypes.html#reference"><i class="fa fa-check"></i><b>6.11</b> Reference</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="tme-interaction-analysis.html"><a href="tme-interaction-analysis.html"><i class="fa fa-check"></i><b>7</b> <strong>TME Interaction analysis</strong></a>
<ul>
<li class="chapter" data-level="7.1" data-path="tme-interaction-analysis.html"><a href="tme-interaction-analysis.html#loading-packages-4"><i class="fa fa-check"></i><b>7.1</b> Loading packages</a></li>
<li class="chapter" data-level="7.2" data-path="tme-interaction-analysis.html"><a href="tme-interaction-analysis.html#downloading-data-for-example-3"><i class="fa fa-check"></i><b>7.2</b> Downloading data for example</a></li>
<li class="chapter" data-level="7.3" data-path="tme-interaction-analysis.html"><a href="tme-interaction-analysis.html#gene-annotation-hgu133plus-2-affaymetrix"><i class="fa fa-check"></i><b>7.3</b> Gene Annotation: HGU133PLUS-2 (Affaymetrix)</a></li>
<li class="chapter" data-level="7.4" data-path="tme-interaction-analysis.html"><a href="tme-interaction-analysis.html#tme-deconvolution-using-cibersort-algorithm"><i class="fa fa-check"></i><b>7.4</b> TME deconvolution using CIBERSORT algorithm</a></li>
<li class="chapter" data-level="7.5" data-path="tme-interaction-analysis.html"><a href="tme-interaction-analysis.html#identifying-tme-patterns"><i class="fa fa-check"></i><b>7.5</b> Identifying TME patterns</a></li>
<li class="chapter" data-level="7.6" data-path="tme-interaction-analysis.html"><a href="tme-interaction-analysis.html#cell-abundance-of-each-cluster"><i class="fa fa-check"></i><b>7.6</b> Cell abundance of each cluster</a></li>
<li class="chapter" data-level="7.7" data-path="tme-interaction-analysis.html"><a href="tme-interaction-analysis.html#deg-analysis-between-tme-subtypes"><i class="fa fa-check"></i><b>7.7</b> DEG analysis between TME subtypes</a></li>
<li class="chapter" data-level="7.8" data-path="tme-interaction-analysis.html"><a href="tme-interaction-analysis.html#identifying-signatures-associated-with-tme-clusters"><i class="fa fa-check"></i><b>7.8</b> Identifying signatures associated with TME clusters</a></li>
<li class="chapter" data-level="7.9" data-path="tme-interaction-analysis.html"><a href="tme-interaction-analysis.html#references-3"><i class="fa fa-check"></i><b>7.9</b> References</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="tumor-ecosystem-analysis.html"><a href="tumor-ecosystem-analysis.html"><i class="fa fa-check"></i><b>8</b> <strong>Tumor ecosystem analysis</strong></a>
<ul>
<li class="chapter" data-level="8.1" data-path="tumor-ecosystem-analysis.html"><a href="tumor-ecosystem-analysis.html#loading-packages-5"><i class="fa fa-check"></i><b>8.1</b> Loading packages</a></li>
<li class="chapter" data-level="8.2" data-path="tumor-ecosystem-analysis.html"><a href="tumor-ecosystem-analysis.html#downloading-data-for-example-4"><i class="fa fa-check"></i><b>8.2</b> Downloading data for example</a></li>
<li class="chapter" data-level="8.3" data-path="tumor-ecosystem-analysis.html"><a href="tumor-ecosystem-analysis.html#gene-annotation-hgu133plus-2-affaymetrix-1"><i class="fa fa-check"></i><b>8.3</b> Gene Annotation: HGU133PLUS-2 (Affaymetrix)</a></li>
<li class="chapter" data-level="8.4" data-path="tumor-ecosystem-analysis.html"><a href="tumor-ecosystem-analysis.html#determine-tme-subtype-of-gastric-cancer-using-tmeclassifier-r-package"><i class="fa fa-check"></i><b>8.4</b> Determine TME subtype of gastric cancer using TMEclassifier R package</a></li>
<li class="chapter" data-level="8.5" data-path="tumor-ecosystem-analysis.html"><a href="tumor-ecosystem-analysis.html#deg-analysis-method1"><i class="fa fa-check"></i><b>8.5</b> DEG analysis: method1</a></li>
<li class="chapter" data-level="8.6" data-path="tumor-ecosystem-analysis.html"><a href="tumor-ecosystem-analysis.html#gsea-analysis-based-on-differential-express-gene-analysis-results"><i class="fa fa-check"></i><b>8.6</b> GSEA analysis based on differential express gene analysis results</a></li>
<li class="chapter" data-level="8.7" data-path="tumor-ecosystem-analysis.html"><a href="tumor-ecosystem-analysis.html#deg-analysis-method2"><i class="fa fa-check"></i><b>8.7</b> DEG analysis: method2</a></li>
<li class="chapter" data-level="8.8" data-path="tumor-ecosystem-analysis.html"><a href="tumor-ecosystem-analysis.html#identifying-signatures-associated-with-tme-clusters-1"><i class="fa fa-check"></i><b>8.8</b> Identifying signatures associated with TME clusters</a></li>
<li class="chapter" data-level="8.9" data-path="tumor-ecosystem-analysis.html"><a href="tumor-ecosystem-analysis.html#references-4"><i class="fa fa-check"></i><b>8.9</b> References</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="tme-and-genomic-interaction.html"><a href="tme-and-genomic-interaction.html"><i class="fa fa-check"></i><b>9</b> <strong>TME and genomic interaction</strong></a>
<ul>
<li class="chapter" data-level="9.1" data-path="tme-and-genomic-interaction.html"><a href="tme-and-genomic-interaction.html#loading-packages-6"><i class="fa fa-check"></i><b>9.1</b> Loading packages</a></li>
<li class="chapter" data-level="9.2" data-path="tme-and-genomic-interaction.html"><a href="tme-and-genomic-interaction.html#genomic-data-prepare"><i class="fa fa-check"></i><b>9.2</b> Genomic data prepare</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="tme-and-genomic-interaction.html"><a href="tme-and-genomic-interaction.html#using-tcgabiolinks-to-download-genomic-data"><i class="fa fa-check"></i><b>9.2.1</b> Using TCGAbiolinks to download genomic data</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="tme-and-genomic-interaction.html"><a href="tme-and-genomic-interaction.html#identifying-mutations-associated-with-tme"><i class="fa fa-check"></i><b>9.3</b> Identifying Mutations Associated with TME</a></li>
<li class="chapter" data-level="9.4" data-path="tme-and-genomic-interaction.html"><a href="tme-and-genomic-interaction.html#oncoprint-of-result"><i class="fa fa-check"></i><b>9.4</b> OncoPrint of result</a></li>
<li class="chapter" data-level="9.5" data-path="tme-and-genomic-interaction.html"><a href="tme-and-genomic-interaction.html#boxplot-of-top-10-mutated-genes"><i class="fa fa-check"></i><b>9.5</b> Boxplot of top 10 mutated genes</a></li>
<li class="chapter" data-level="9.6" data-path="tme-and-genomic-interaction.html"><a href="tme-and-genomic-interaction.html#other-methods-to-obtaind-genomic-data"><i class="fa fa-check"></i><b>9.6</b> Other methods to obtaind genomic data</a>
<ul>
<li class="chapter" data-level="9.6.1" data-path="tme-and-genomic-interaction.html"><a href="tme-and-genomic-interaction.html#using-tcgamutations"><i class="fa fa-check"></i><b>9.6.1</b> Using TCGAmutations</a></li>
<li class="chapter" data-level="9.6.2" data-path="tme-and-genomic-interaction.html"><a href="tme-and-genomic-interaction.html#using-maftools"><i class="fa fa-check"></i><b>9.6.2</b> Using maftools</a></li>
</ul></li>
<li class="chapter" data-level="9.7" data-path="tme-and-genomic-interaction.html"><a href="tme-and-genomic-interaction.html#references-5"><i class="fa fa-check"></i><b>9.7</b> References</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="tme-modeling.html"><a href="tme-modeling.html"><i class="fa fa-check"></i><b>10</b> <strong>TME Modeling</strong></a>
<ul>
<li class="chapter" data-level="10.1" data-path="tme-modeling.html"><a href="tme-modeling.html#loading-packages-7"><i class="fa fa-check"></i><b>10.1</b> Loading packages</a></li>
<li class="chapter" data-level="10.2" data-path="tme-modeling.html"><a href="tme-modeling.html#data-prepare"><i class="fa fa-check"></i><b>10.2</b> Data prepare</a></li>
<li class="chapter" data-level="10.3" data-path="tme-modeling.html"><a href="tme-modeling.html#input-data-overall-survival-prepare"><i class="fa fa-check"></i><b>10.3</b> Input data (overall survival) prepare</a></li>
<li class="chapter" data-level="10.4" data-path="tme-modeling.html"><a href="tme-modeling.html#constructing-survival-prediction-models"><i class="fa fa-check"></i><b>10.4</b> Constructing survival prediction models</a></li>
<li class="chapter" data-level="10.5" data-path="tme-modeling.html"><a href="tme-modeling.html#input-data-response-prepare"><i class="fa fa-check"></i><b>10.5</b> Input data (Response) prepare</a></li>
<li class="chapter" data-level="10.6" data-path="tme-modeling.html"><a href="tme-modeling.html#constructing-prediction-models-for-response"><i class="fa fa-check"></i><b>10.6</b> Constructing prediction models for response</a></li>
<li class="chapter" data-level="10.7" data-path="tme-modeling.html"><a href="tme-modeling.html#references-6"><i class="fa fa-check"></i><b>10.7</b> References</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="references-7.html"><a href="references-7.html"><i class="fa fa-check"></i><b>11</b> <strong>References</strong></a>
<ul>
<li class="chapter" data-level="11.1" data-path="references-7.html"><a href="references-7.html#tme-deconvolution-1"><i class="fa fa-check"></i><b>11.1</b> TME deconvolution</a></li>
<li class="chapter" data-level="11.2" data-path="references-7.html"><a href="references-7.html#tme-signatures"><i class="fa fa-check"></i><b>11.2</b> TME Signatures</a></li>
<li class="chapter" data-level="11.3" data-path="references-7.html"><a href="references-7.html#data-sets"><i class="fa fa-check"></i><b>11.3</b> Data sets</a></li>
<li class="chapter" data-level="11.4" data-path="references-7.html"><a href="references-7.html#others"><i class="fa fa-check"></i><b>11.4</b> Others</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/DongqiangZeng0808" target="blank">Welcome to my GitHub</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">IOBR (Immuno-Oncology Biological Research)</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="signature-score-calculation" class="section level1 hasAnchor" number="4">
<h1><span class="header-section-number">Chapter 4</span> <strong>Signature Score Calculation</strong><a href="signature-score-calculation.html#signature-score-calculation" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="loading-packages-1" class="section level2 hasAnchor" number="4.1">
<h2><span class="header-section-number">4.1</span> Loading packages<a href="signature-score-calculation.html#loading-packages-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Load the IOBR package in your R session after the installation is complete:</p>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb53-1"><a href="signature-score-calculation.html#cb53-1" tabindex="-1"></a><span class="fu">library</span>(IOBR)</span>
<span id="cb53-2"><a href="signature-score-calculation.html#cb53-2" tabindex="-1"></a><span class="fu">library</span>(survminer)</span>
<span id="cb53-3"><a href="signature-score-calculation.html#cb53-3" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code></pre></div>
</div>
<div id="downloading-data-for-example" class="section level2 hasAnchor" number="4.2">
<h2><span class="header-section-number">4.2</span> Downloading data for example<a href="signature-score-calculation.html#downloading-data-for-example" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Obtaining data set from GEO <a href="https://pubmed.ncbi.nlm.nih.gov/25894828/">Gastric cancer: GSE62254</a> using <code>GEOquery</code> R package.</p>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="signature-score-calculation.html#cb54-1" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;GEOquery&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))  BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;GEOquery&quot;</span>)</span>
<span id="cb54-2"><a href="signature-score-calculation.html#cb54-2" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;GEOquery&quot;</span>)</span>
<span id="cb54-3"><a href="signature-score-calculation.html#cb54-3" tabindex="-1"></a><span class="co"># </span><span class="al">NOTE</span><span class="co">: This process may take a few minutes which depends on the internet connection speed. Please wait for its completion.</span></span>
<span id="cb54-4"><a href="signature-score-calculation.html#cb54-4" tabindex="-1"></a>eset_geo <span class="ot">&lt;-</span> <span class="fu">getGEO</span>(<span class="at">GEO =</span> <span class="st">&quot;GSE62254&quot;</span>, <span class="at">getGPL  =</span> F, <span class="at">destdir =</span> <span class="st">&quot;./&quot;</span>)</span>
<span id="cb54-5"><a href="signature-score-calculation.html#cb54-5" tabindex="-1"></a>eset    <span class="ot">&lt;-</span>eset_geo[[<span class="dv">1</span>]]</span>
<span id="cb54-6"><a href="signature-score-calculation.html#cb54-6" tabindex="-1"></a>eset    <span class="ot">&lt;-</span><span class="fu">exprs</span>(eset)</span>
<span id="cb54-7"><a href="signature-score-calculation.html#cb54-7" tabindex="-1"></a>eset[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span></code></pre></div>
<pre><code>##           GSM1523727 GSM1523728 GSM1523729 GSM1523744 GSM1523745
## 1007_s_at  3.2176645  3.0624323  3.0279131   2.921683  2.8456013
## 1053_at    2.4050109  2.4394879  2.2442708   2.345916  2.4328582
## 117_at     1.4933412  1.8067380  1.5959665   1.839822  1.8326058
## 121_at     2.1965561  2.2812181  2.1865556   2.258599  2.1874363
## 1255_g_at  0.8698382  0.9502466  0.8125414   1.012860  0.9441993</code></pre>
<p>Annotation of genes in the expression matrix and removal of duplicate genes.</p>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="signature-score-calculation.html#cb56-1" tabindex="-1"></a><span class="co"># Load the annotation file `anno_hug133plus2` in IOBR.</span></span>
<span id="cb56-2"><a href="signature-score-calculation.html#cb56-2" tabindex="-1"></a><span class="fu">head</span>(anno_hug133plus2)</span></code></pre></div>
<pre><code>## # A tibble: 6 √ó 2
##   probe_id  symbol 
##   &lt;fct&gt;     &lt;fct&gt;  
## 1 1007_s_at MIR4640
## 2 1053_at   RFC2   
## 3 117_at    HSPA6  
## 4 121_at    PAX8   
## 5 1255_g_at GUCA1A 
## 6 1294_at   MIR5193</code></pre>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="signature-score-calculation.html#cb58-1" tabindex="-1"></a><span class="co"># Conduct gene annotation using `anno_hug133plus2` file; If identical gene symbols exists, these genes would be ordered by the mean expression levels. The gene symbol with highest mean expression level is selected and remove others. </span></span>
<span id="cb58-2"><a href="signature-score-calculation.html#cb58-2" tabindex="-1"></a></span>
<span id="cb58-3"><a href="signature-score-calculation.html#cb58-3" tabindex="-1"></a>eset<span class="ot">&lt;-</span><span class="fu">anno_eset</span>(<span class="at">eset       =</span> eset,</span>
<span id="cb58-4"><a href="signature-score-calculation.html#cb58-4" tabindex="-1"></a>                <span class="at">annotation =</span> anno_hug133plus2,</span>
<span id="cb58-5"><a href="signature-score-calculation.html#cb58-5" tabindex="-1"></a>                <span class="at">symbol     =</span> <span class="st">&quot;symbol&quot;</span>,</span>
<span id="cb58-6"><a href="signature-score-calculation.html#cb58-6" tabindex="-1"></a>                <span class="at">probe      =</span> <span class="st">&quot;probe_id&quot;</span>,</span>
<span id="cb58-7"><a href="signature-score-calculation.html#cb58-7" tabindex="-1"></a>                <span class="at">method     =</span> <span class="st">&quot;mean&quot;</span>)</span>
<span id="cb58-8"><a href="signature-score-calculation.html#cb58-8" tabindex="-1"></a>eset[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]</span></code></pre></div>
<pre><code>##              GSM1523727 GSM1523728 GSM1523729
## SH3KBP1        4.327974   4.316195   4.351425
## RPL41          4.246149   4.246808   4.257940
## EEF1A1         4.293762   4.291038   4.262199
## COX2           4.250288   4.283714   4.270508
## LOC101928826   4.219303   4.219670   4.213252</code></pre>
</div>
<div id="signature-score-estimation" class="section level2 hasAnchor" number="4.3">
<h2><span class="header-section-number">4.3</span> Signature score estimation<a href="signature-score-calculation.html#signature-score-estimation" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="signature-collection-of-iobr" class="section level3 hasAnchor" number="4.3.1">
<h3><span class="header-section-number">4.3.1</span> Signature collection of IOBR<a href="signature-score-calculation.html#signature-collection-of-iobr" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="signature-score-calculation.html#cb60-1" tabindex="-1"></a><span class="co"># Return available parameter options of signature estimation.</span></span>
<span id="cb60-2"><a href="signature-score-calculation.html#cb60-2" tabindex="-1"></a>signature_score_calculation_methods</span></code></pre></div>
<pre><code>##           PCA        ssGSEA       z-score   Integration 
##         &quot;pca&quot;      &quot;ssgsea&quot;      &quot;zscore&quot; &quot;integration&quot;</code></pre>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="signature-score-calculation.html#cb62-1" tabindex="-1"></a><span class="co">#TME associated signatures</span></span>
<span id="cb62-2"><a href="signature-score-calculation.html#cb62-2" tabindex="-1"></a><span class="fu">names</span>(signature_tme)[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>]</span></code></pre></div>
<pre><code>##  [1] &quot;CD_8_T_effector&quot;            &quot;DDR&quot;                       
##  [3] &quot;APM&quot;                        &quot;Immune_Checkpoint&quot;         
##  [5] &quot;CellCycle_Reg&quot;              &quot;Pan_F_TBRs&quot;                
##  [7] &quot;Histones&quot;                   &quot;EMT1&quot;                      
##  [9] &quot;EMT2&quot;                       &quot;EMT3&quot;                      
## [11] &quot;WNT_target&quot;                 &quot;FGFR3_related&quot;             
## [13] &quot;Cell_cycle&quot;                 &quot;Mismatch_Repair&quot;           
## [15] &quot;Homologous_recombination&quot;   &quot;Nucleotide_excision_repair&quot;
## [17] &quot;DNA_replication&quot;            &quot;Base_excision_repair&quot;      
## [19] &quot;TMEscoreA_CIR&quot;              &quot;TMEscoreB_CIR&quot;</code></pre>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="signature-score-calculation.html#cb64-1" tabindex="-1"></a><span class="co">#Metabolism related signatures</span></span>
<span id="cb64-2"><a href="signature-score-calculation.html#cb64-2" tabindex="-1"></a><span class="fu">names</span>(signature_metabolism)[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>]</span></code></pre></div>
<pre><code>##  [1] &quot;Cardiolipin_Metabolism&quot;                    
##  [2] &quot;Cardiolipin_Biosynthesis&quot;                  
##  [3] &quot;Cholesterol_Biosynthesis&quot;                  
##  [4] &quot;Citric_Acid_Cycle&quot;                         
##  [5] &quot;Cyclooxygenase_Arachidonic_Acid_Metabolism&quot;
##  [6] &quot;Prostaglandin_Biosynthesis&quot;                
##  [7] &quot;Purine_Biosynthesis&quot;                       
##  [8] &quot;Pyrimidine_Biosynthesis&quot;                   
##  [9] &quot;Dopamine_Biosynthesis&quot;                     
## [10] &quot;Epinephrine_Biosynthesis&quot;                  
## [11] &quot;Norepinephrine_Biosynthesis&quot;               
## [12] &quot;Fatty_Acid_Degradation&quot;                    
## [13] &quot;Fatty_Acid_Elongation&quot;                     
## [14] &quot;Fatty_Acid_Biosynthesis&quot;                   
## [15] &quot;Folate_One_Carbon_Metabolism&quot;              
## [16] &quot;Folate_biosynthesis&quot;                       
## [17] &quot;Gluconeogenesis&quot;                           
## [18] &quot;Glycolysis&quot;                                
## [19] &quot;Glycogen_Biosynthesis&quot;                     
## [20] &quot;Glycogen_Degradation&quot;</code></pre>
<p>Signatures associated with basic biomedical research, such as m6A, TLS, ferroptosis and exosomes.</p>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="signature-score-calculation.html#cb66-1" tabindex="-1"></a><span class="fu">names</span>(signature_tumor)</span></code></pre></div>
<pre><code>##  [1] &quot;Nature_metabolism_Hypoxia&quot;                
##  [2] &quot;Winter_hypoxia_signature&quot;                 
##  [3] &quot;Hu_hypoxia_signature&quot;                     
##  [4] &quot;Molecular_Cancer_m6A&quot;                     
##  [5] &quot;MT_exosome&quot;                               
##  [6] &quot;SR_exosome&quot;                               
##  [7] &quot;Positive_regulation_of_exosomal_secretion&quot;
##  [8] &quot;Negative_regulation_of_exosomal_secretion&quot;
##  [9] &quot;Exosomal_secretion&quot;                       
## [10] &quot;Exosome_assembly&quot;                         
## [11] &quot;Extracellular_vesicle_biogenesis&quot;         
## [12] &quot;MC_Review_Exosome1&quot;                       
## [13] &quot;MC_Review_Exosome2&quot;                       
## [14] &quot;CMLS_Review_Exosome&quot;                      
## [15] &quot;Ferroptosis&quot;                              
## [16] &quot;EV_Cell_2020&quot;</code></pre>
<p><code>signature_collection</code> including all aforementioned signatures</p>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="signature-score-calculation.html#cb68-1" tabindex="-1"></a><span class="fu">names</span>(signature_collection)[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>]</span></code></pre></div>
<pre><code>##  [1] &quot;CD_8_T_effector&quot;            &quot;DDR&quot;                       
##  [3] &quot;APM&quot;                        &quot;Immune_Checkpoint&quot;         
##  [5] &quot;CellCycle_Reg&quot;              &quot;Pan_F_TBRs&quot;                
##  [7] &quot;Histones&quot;                   &quot;EMT1&quot;                      
##  [9] &quot;EMT2&quot;                       &quot;EMT3&quot;                      
## [11] &quot;WNT_target&quot;                 &quot;FGFR3_related&quot;             
## [13] &quot;Cell_cycle&quot;                 &quot;Mismatch_Repair&quot;           
## [15] &quot;Homologous_recombination&quot;   &quot;Nucleotide_excision_repair&quot;
## [17] &quot;DNA_replication&quot;            &quot;Base_excision_repair&quot;      
## [19] &quot;TMEscoreA_CIR&quot;              &quot;TMEscoreB_CIR&quot;</code></pre>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="signature-score-calculation.html#cb70-1" tabindex="-1"></a><span class="co">#citation of signatures</span></span>
<span id="cb70-2"><a href="signature-score-calculation.html#cb70-2" tabindex="-1"></a>signature_collection_citation[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>, ]</span></code></pre></div>
<pre><code>## # A tibble: 20 √ó 6
##    Signatures                 `Published year` Journal         Title PMID  DOI  
##    &lt;chr&gt;                                 &lt;dbl&gt; &lt;chr&gt;           &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;
##  1 CD_8_T_effector                        2018 Nature          TGFŒ≤‚Ä¶ 2944‚Ä¶ 10.1‚Ä¶
##  2 DDR                                    2018 Nature          TGFŒ≤‚Ä¶ 2944‚Ä¶ 10.1‚Ä¶
##  3 APM                                    2018 Nature          TGFŒ≤‚Ä¶ 2944‚Ä¶ 10.1‚Ä¶
##  4 Immune_Checkpoint                      2018 Nature          TGFŒ≤‚Ä¶ 2944‚Ä¶ 10.1‚Ä¶
##  5 CellCycle_Reg                          2018 Nature          TGFŒ≤‚Ä¶ 2944‚Ä¶ 10.1‚Ä¶
##  6 Pan_F_TBRs                             2018 Nature          TGFŒ≤‚Ä¶ 2944‚Ä¶ 10.1‚Ä¶
##  7 Histones                               2018 Nature          TGFŒ≤‚Ä¶ 2944‚Ä¶ 10.1‚Ä¶
##  8 EMT1                                   2018 Nature          TGFŒ≤‚Ä¶ 2944‚Ä¶ 10.1‚Ä¶
##  9 EMT2                                   2018 Nature          TGFŒ≤‚Ä¶ 2944‚Ä¶ 10.1‚Ä¶
## 10 EMT3                                   2018 Nature          TGFŒ≤‚Ä¶ 2944‚Ä¶ 10.1‚Ä¶
## 11 WNT_target                             2018 Nature          TGFŒ≤‚Ä¶ 2944‚Ä¶ 10.1‚Ä¶
## 12 FGFR3_related                          2018 Nature          TGFŒ≤‚Ä¶ 2944‚Ä¶ 10.1‚Ä¶
## 13 Cell_cycle                             2018 Nature          TGFŒ≤‚Ä¶ 2944‚Ä¶ 10.1‚Ä¶
## 14 Mismatch_Repair                        2018 Nature          TGFŒ≤‚Ä¶ 2944‚Ä¶ 10.1‚Ä¶
## 15 Homologous_recombination               2018 Nature          TGFŒ≤‚Ä¶ 2944‚Ä¶ 10.1‚Ä¶
## 16 Nucleotide_excision_repair             2018 Nature          TGFŒ≤‚Ä¶ 2944‚Ä¶ 10.1‚Ä¶
## 17 DNA_replication                        2018 Nature          TGFŒ≤‚Ä¶ 2944‚Ä¶ 10.1‚Ä¶
## 18 Base_excision_repair                   2018 Nature          TGFŒ≤‚Ä¶ 2944‚Ä¶ 10.1‚Ä¶
## 19 TMEscoreA_CIR                          2019 Cancer Immunol‚Ä¶ Tumo‚Ä¶ 3084‚Ä¶ 10.1‚Ä¶
## 20 TMEscoreB_CIR                          2019 Cancer Immunol‚Ä¶ Tumo‚Ä¶ 3084‚Ä¶ 10.1‚Ä¶</code></pre>
<p>The evaluation of signature scores involved three methodologies: Single-sample Gene Set Enrichment Analysis (ssGSEA), Principal Component Analysis (PCA), and Z-score.</p>
</div>
</div>
<div id="estimation-of-signature-using-pca-method" class="section level2 hasAnchor" number="4.4">
<h2><span class="header-section-number">4.4</span> Estimation of signature using PCA method<a href="signature-score-calculation.html#estimation-of-signature-using-pca-method" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The PCA method is ideal for gene sets with co-expression. Heatmaps and correlation matrices can be used to determine if co-expression is present in the applicable gene set.</p>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="signature-score-calculation.html#cb72-1" tabindex="-1"></a>sig_tme<span class="ot">&lt;-</span><span class="fu">calculate_sig_score</span>(<span class="at">pdata           =</span> <span class="cn">NULL</span>,</span>
<span id="cb72-2"><a href="signature-score-calculation.html#cb72-2" tabindex="-1"></a>                             <span class="at">eset            =</span> eset,</span>
<span id="cb72-3"><a href="signature-score-calculation.html#cb72-3" tabindex="-1"></a>                             <span class="at">signature       =</span> signature_collection,</span>
<span id="cb72-4"><a href="signature-score-calculation.html#cb72-4" tabindex="-1"></a>                             <span class="at">method          =</span> <span class="st">&quot;pca&quot;</span>,</span>
<span id="cb72-5"><a href="signature-score-calculation.html#cb72-5" tabindex="-1"></a>                             <span class="at">mini_gene_count =</span> <span class="dv">2</span>)</span>
<span id="cb72-6"><a href="signature-score-calculation.html#cb72-6" tabindex="-1"></a></span>
<span id="cb72-7"><a href="signature-score-calculation.html#cb72-7" tabindex="-1"></a>sig_tme <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">column_to_rownames</span>(sig_tme, <span class="at">var =</span> <span class="st">&quot;ID&quot;</span>))</span>
<span id="cb72-8"><a href="signature-score-calculation.html#cb72-8" tabindex="-1"></a>sig_tme[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]</span></code></pre></div>
<pre><code>##                   GSM1523727 GSM1523728 GSM1523729
## CD_8_T_effector   -2.5513794  0.7789141 -2.1770675
## DDR               -0.8747614  0.7425162 -1.3272054
## APM                1.1098368  2.1988688 -0.9516419
## Immune_Checkpoint -2.3701787  0.9455120 -1.4844104
## CellCycle_Reg      0.1063358  0.7583302 -0.3649795</code></pre>
</div>
<div id="estimated-using-the-ssgsea-methodology" class="section level2 hasAnchor" number="4.5">
<h2><span class="header-section-number">4.5</span> Estimated using the ssGSEA methodology<a href="signature-score-calculation.html#estimated-using-the-ssgsea-methodology" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>This method is appropriate for gene sets that contain a large number of genes (&gt; 30 genes), such as those of <a href="https://www.gsea-msigdb.org/gsea/msigdb">GO, KEGG, REACTOME gene sets</a>.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-10"></span>
<img src="fig/gsea.png" alt="Gene sets of MSigDb" width="95%" />
<p class="caption">
Figure 4.1: Gene sets of MSigDb
</p>
</div>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="signature-score-calculation.html#cb74-1" tabindex="-1"></a>sig_tme<span class="ot">&lt;-</span><span class="fu">calculate_sig_score</span>(<span class="at">pdata           =</span> <span class="cn">NULL</span>,</span>
<span id="cb74-2"><a href="signature-score-calculation.html#cb74-2" tabindex="-1"></a>                             <span class="at">eset            =</span> eset,</span>
<span id="cb74-3"><a href="signature-score-calculation.html#cb74-3" tabindex="-1"></a>                             <span class="at">signature       =</span> go_bp,</span>
<span id="cb74-4"><a href="signature-score-calculation.html#cb74-4" tabindex="-1"></a>                             <span class="at">method          =</span> <span class="st">&quot;ssgsea&quot;</span>,</span>
<span id="cb74-5"><a href="signature-score-calculation.html#cb74-5" tabindex="-1"></a>                             <span class="at">mini_gene_count =</span> <span class="dv">2</span>)</span></code></pre></div>
</div>
<div id="calculated-using-the-z-score-function." class="section level2 hasAnchor" number="4.6">
<h2><span class="header-section-number">4.6</span> Calculated using the z-score function.<a href="signature-score-calculation.html#calculated-using-the-z-score-function." class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb75-1"><a href="signature-score-calculation.html#cb75-1" tabindex="-1"></a>sig_tme<span class="ot">&lt;-</span><span class="fu">calculate_sig_score</span>(<span class="at">pdata           =</span> <span class="cn">NULL</span>,</span>
<span id="cb75-2"><a href="signature-score-calculation.html#cb75-2" tabindex="-1"></a>                             <span class="at">eset            =</span> eset,</span>
<span id="cb75-3"><a href="signature-score-calculation.html#cb75-3" tabindex="-1"></a>                             <span class="at">signature       =</span> signature_collection,</span>
<span id="cb75-4"><a href="signature-score-calculation.html#cb75-4" tabindex="-1"></a>                             <span class="at">method          =</span> <span class="st">&quot;zscore&quot;</span>,</span>
<span id="cb75-5"><a href="signature-score-calculation.html#cb75-5" tabindex="-1"></a>                             <span class="at">mini_gene_count =</span> <span class="dv">2</span>)</span></code></pre></div>
</div>
<div id="calculated-using-all-three-methods-at-the-same-time" class="section level2 hasAnchor" number="4.7">
<h2><span class="header-section-number">4.7</span> Calculated using all three methods at the same time<a href="signature-score-calculation.html#calculated-using-all-three-methods-at-the-same-time" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb76-1"><a href="signature-score-calculation.html#cb76-1" tabindex="-1"></a>sig_tme<span class="ot">&lt;-</span><span class="fu">calculate_sig_score</span>(<span class="at">pdata           =</span> <span class="cn">NULL</span>,</span>
<span id="cb76-2"><a href="signature-score-calculation.html#cb76-2" tabindex="-1"></a>                             <span class="at">eset            =</span> eset,</span>
<span id="cb76-3"><a href="signature-score-calculation.html#cb76-3" tabindex="-1"></a>                             <span class="at">signature       =</span> signature_collection,</span>
<span id="cb76-4"><a href="signature-score-calculation.html#cb76-4" tabindex="-1"></a>                             <span class="at">method          =</span> <span class="st">&quot;integration&quot;</span>,</span>
<span id="cb76-5"><a href="signature-score-calculation.html#cb76-5" tabindex="-1"></a>                             <span class="at">mini_gene_count =</span> <span class="dv">2</span>)</span></code></pre></div>
<p>The same SIGNATURE in this case will be scored using all three methods simultaneously.</p>
<div class="sourceCode" id="cb77"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb77-1"><a href="signature-score-calculation.html#cb77-1" tabindex="-1"></a><span class="fu">colnames</span>(sig_tme)[<span class="fu">grep</span>(<span class="fu">colnames</span>(sig_tme), <span class="at">pattern =</span> <span class="st">&quot;CD_8_T_effector&quot;</span>)]</span></code></pre></div>
<p>The <code>select_method()</code> function allows the user to extract data using various methods.</p>
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb78-1"><a href="signature-score-calculation.html#cb78-1" tabindex="-1"></a>sig_tme_pca <span class="ot">&lt;-</span> <span class="fu">select_method</span>(<span class="at">data =</span> sig_tme, <span class="at">method =</span> <span class="st">&quot;pca&quot;</span>)</span>
<span id="cb78-2"><a href="signature-score-calculation.html#cb78-2" tabindex="-1"></a><span class="fu">colnames</span>(sig_tme_pca)[<span class="fu">grep</span>(<span class="fu">colnames</span>(sig_tme_pca), <span class="at">pattern =</span> <span class="st">&quot;CD_8_T_effector&quot;</span>)]</span></code></pre></div>
</div>
<div id="how-to-customise-the-signature-gene-list-for-calculate_signature_score" class="section level2 hasAnchor" number="4.8">
<h2><span class="header-section-number">4.8</span> How to customise the signature gene list for <code>calculate_signature_score</code><a href="signature-score-calculation.html#how-to-customise-the-signature-gene-list-for-calculate_signature_score" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="method-1-use-excel-for-storage-and-construction" class="section level3 hasAnchor" number="4.8.1">
<h3><span class="header-section-number">4.8.1</span> Method-1: Use excel for storage and construction<a href="signature-score-calculation.html#method-1-use-excel-for-storage-and-construction" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Users can collect gene signatures using either an <code>Excel</code> or <code>CSV</code> file. The format should have the name of the signature in the first row, followed by the genes contained in each signature from the second row onwards. Once imported, the function <code>format_signature</code> can be used to transform the data into a gene list of signatures required for <code>calculate_signature_score</code>. To import the file into R, users can use the functions read.csv or read_excel. It is important to note here that the user needs to use the longest signature as a criterion and then replace all the vacant grids using NA, otherwise an error may be reported when reading into R.</p>
<p>Here we provide a sample data <code>sig_excel</code>, please refer to this format to construct the required csv or excel files.</p>
<div class="sourceCode" id="cb79"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb79-1"><a href="signature-score-calculation.html#cb79-1" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sig_excel&quot;</span>, <span class="at">package =</span> <span class="st">&quot;IOBR&quot;</span>)</span>
<span id="cb79-2"><a href="signature-score-calculation.html#cb79-2" tabindex="-1"></a>sig <span class="ot">&lt;-</span> <span class="fu">format_signatures</span>(sig_excel)</span>
<span id="cb79-3"><a href="signature-score-calculation.html#cb79-3" tabindex="-1"></a><span class="fu">print</span>(sig[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>])</span></code></pre></div>
<pre><code>## $Tcell_co_inhibitors
##  [1] &quot;ADORA2A&quot;  &quot;BTLA&quot;     &quot;BTN2A2&quot;   &quot;BTN3A1&quot;   &quot;BTN3A2&quot;   &quot;BTNL2&quot;   
##  [7] &quot;C10orf54&quot; &quot;CSF1R&quot;    &quot;HAVCR2&quot;   &quot;IDO1&quot;     &quot;IL10&quot;     &quot;IL10RB&quot;  
## [13] &quot;KDR&quot;      &quot;KIR2DL1&quot;  &quot;SLAMF7&quot;   &quot;TGFB1&quot;    &quot;TIGIT&quot;    &quot;VRCN1&quot;   
## [19] &quot;VTCN1&quot;    &quot;CD247&quot;    &quot;CTLA4&quot;    &quot;CD160&quot;    &quot;CD244&quot;    &quot;CD274&quot;   
## [25] &quot;CD276&quot;    &quot;CD48&quot;     &quot;CD96&quot;     &quot;KIR2DL2&quot;  &quot;KIR2DL3&quot;  &quot;LAG3&quot;    
## [31] &quot;LAIR1&quot;    &quot;LGALS9&quot;   &quot;PVRL2&quot;    &quot;PDCD1&quot;    &quot;PDCD1LG2&quot;
## 
## $Tcell_co_stimuiations
##  [1] &quot;BTNL8&quot;     &quot;CD226&quot;     &quot;CD27&quot;      &quot;CD28&quot;      &quot;CD40&quot;      &quot;CD58&quot;     
##  [7] &quot;CD70&quot;      &quot;SLAMF1&quot;    &quot;TMIGD2&quot;    &quot;TNFRSF13B&quot; &quot;TNFRSF13C&quot; &quot;TNFRSF14&quot; 
## [13] &quot;TNFRSF4&quot;   &quot;TNFRSF8&quot;   &quot;TNFSF8&quot;    &quot;TNFSF9&quot;    &quot;ENTPD1&quot;    &quot;NT5E&quot;     
## [19] &quot;ICOS&quot;      &quot;TNFSF4&quot;    &quot;TNFSF15&quot;   &quot;CD80&quot;      &quot;CD86&quot;      &quot;EGFR&quot;     
## [25] &quot;HAVCR1&quot;    &quot;TNFSF18&quot;   &quot;ICOSLG&quot;    &quot;TNFSF13B&quot;  &quot;TNFRSF9&quot;   &quot;TNFSF13&quot;  
## 
## $Tcell_function
## [1] &quot;CD3E&quot;  &quot;CD4&quot;   &quot;CD8B&quot;  &quot;FOXP3&quot; &quot;GZMB&quot;  &quot;PRF1&quot;  &quot;TBX21&quot; &quot;IL2RA&quot; &quot;IKZF2&quot;
## 
## $Tcell_checkpoint
##  [1] &quot;CD274&quot;    &quot;CTLA4&quot;    &quot;LAG3&quot;     &quot;TIM3&quot;     &quot;TNFRSF9&quot;  &quot;TIGIT&quot;   
##  [7] &quot;CD226&quot;    &quot;CD7&quot;      &quot;GZMB&quot;     &quot;PRF1&quot;     &quot;TNFRSF18&quot; &quot;TNFRSF4&quot; 
## [13] &quot;HAVCR2&quot;   &quot;NLG1&quot;     &quot;CD4&quot;      &quot;CD8A&quot;     &quot;CD8B&quot;     &quot;FOXP3&quot;   
## [19] &quot;IL2&quot;      &quot;CXCL8&quot;    &quot;PDCD1&quot;    &quot;IFNG&quot;    
## 
## $Teffctore_score
## [1] &quot;CD8A&quot;   &quot;CXCL10&quot; &quot;CXCL9&quot;  &quot;GZMA&quot;   &quot;GZMB&quot;   &quot;IFNG&quot;   &quot;PRF1&quot;   &quot;TBX21&quot;</code></pre>
<p>For simple structures or when the number of signatures to be added is relatively small, the following two methods can also be used.</p>
</div>
<div id="method-2-build-the-list-structure-directly" class="section level3 hasAnchor" number="4.8.2">
<h3><span class="header-section-number">4.8.2</span> Method-2: Build the list structure directly<a href="signature-score-calculation.html#method-2-build-the-list-structure-directly" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb81"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb81-1"><a href="signature-score-calculation.html#cb81-1" tabindex="-1"></a>sig <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="st">&quot;CD8&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;CD8A&quot;</span>,  <span class="st">&quot;CXCL10&quot;</span>, <span class="st">&quot;CXCL9&quot;</span>,  <span class="st">&quot;GZMA&quot;</span>,   <span class="st">&quot;GZMB&quot;</span>,   <span class="st">&quot;IFNG&quot;</span>,   <span class="st">&quot;PRF1&quot;</span>,   <span class="st">&quot;TBX21&quot;</span>),</span>
<span id="cb81-2"><a href="signature-score-calculation.html#cb81-2" tabindex="-1"></a>            <span class="st">&quot;ICB&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;CD274&quot;</span>,   <span class="st">&quot;PDCD1LG2&quot;</span>, <span class="st">&quot;CTLA4&quot;</span>,    <span class="st">&quot;PDCD1&quot;</span>,    <span class="st">&quot;LAG3&quot;</span>,     <span class="st">&quot;HAVCR2&quot;</span>,   <span class="st">&quot;TIGIT&quot;</span> ))</span>
<span id="cb81-3"><a href="signature-score-calculation.html#cb81-3" tabindex="-1"></a>sig</span></code></pre></div>
<pre><code>## $CD8
## [1] &quot;CD8A&quot;   &quot;CXCL10&quot; &quot;CXCL9&quot;  &quot;GZMA&quot;   &quot;GZMB&quot;   &quot;IFNG&quot;   &quot;PRF1&quot;   &quot;TBX21&quot; 
## 
## $ICB
## [1] &quot;CD274&quot;    &quot;PDCD1LG2&quot; &quot;CTLA4&quot;    &quot;PDCD1&quot;    &quot;LAG3&quot;     &quot;HAVCR2&quot;   &quot;TIGIT&quot;</code></pre>
</div>
<div id="method3-add-the-new-signature-to-the-existing-gene-list" class="section level3 hasAnchor" number="4.8.3">
<h3><span class="header-section-number">4.8.3</span> Method3: Add the new signature to the existing gene list<a href="signature-score-calculation.html#method3-add-the-new-signature-to-the-existing-gene-list" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb83"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb83-1"><a href="signature-score-calculation.html#cb83-1" tabindex="-1"></a>sig<span class="ot">&lt;-</span> signature_tumor</span>
<span id="cb83-2"><a href="signature-score-calculation.html#cb83-2" tabindex="-1"></a>sig<span class="sc">$</span>CD8 <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;CD8A&quot;</span>,  <span class="st">&quot;CXCL10&quot;</span>, <span class="st">&quot;CXCL9&quot;</span>,  <span class="st">&quot;GZMA&quot;</span>,   <span class="st">&quot;GZMB&quot;</span>,   <span class="st">&quot;IFNG&quot;</span>,   <span class="st">&quot;PRF1&quot;</span>,   <span class="st">&quot;TBX21&quot;</span>)</span>
<span id="cb83-3"><a href="signature-score-calculation.html#cb83-3" tabindex="-1"></a>sig</span></code></pre></div>
<pre><code>## $Nature_metabolism_Hypoxia
##  [1] &quot;ACOT7&quot;  &quot;SLC2A1&quot; &quot;ALDOA&quot;  &quot;CDKN3&quot;  &quot;ENO1&quot;   &quot;LDHA&quot;   &quot;MIF&quot;    &quot;MRPS17&quot;
##  [9] &quot;NDRG1&quot;  &quot;P4HA1&quot;  &quot;PGAM1&quot;  &quot;TPI1&quot;   &quot;TUBB6&quot;  &quot;VEGFA&quot;  &quot;ADM&quot;   
## 
## $Winter_hypoxia_signature
## [1] &quot;VEGF&quot;  &quot;GLUT1&quot; &quot;PDK-1&quot; &quot;EN01&quot;  &quot;HK2&quot;   &quot;CA9&quot;   &quot;AK3&quot;   &quot;CCNG2&quot; &quot;PFKB3&quot;
## 
## $Hu_hypoxia_signature
##  [1] &quot;FABP5&quot;     &quot;UCHL1&quot;     &quot;GAL&quot;       &quot;PLODDDIT4&quot; &quot;VEGF&quot;      &quot;ADM&quot;      
##  [7] &quot;ANGPTL4&quot;   &quot;NDRG1&quot;     &quot;NP&quot;        &quot;SLC16A3&quot;   &quot;C14ORF58&quot;  &quot;RRAGD&quot;    
## 
## $Molecular_Cancer_m6A
##  [1] &quot;METTL3&quot;    &quot;METTL14&quot;   &quot;RBM15&quot;     &quot;RBM15B&quot;    &quot;WTAP&quot;      &quot;KIAA1429&quot; 
##  [7] &quot;CBLL1&quot;     &quot;ZC3H13&quot;    &quot;ALKBH5&quot;    &quot;FTO&quot;       &quot;YTHDC1&quot;    &quot;YTHDC2&quot;   
## [13] &quot;YTHDF1&quot;    &quot;YTHDF2&quot;    &quot;YTHDF3&quot;    &quot;IGF2BP1&quot;   &quot;HNRNPA2B1&quot; &quot;HNRNPC&quot;   
## [19] &quot;FMR1&quot;      &quot;LRPPRC&quot;    &quot;ELAVL1&quot;   
## 
## $MT_exosome
##  [1] &quot;YWHAG&quot;  &quot;YWHAQ&quot;  &quot;CLTC&quot;   &quot;NCKAP1&quot; &quot;CFL1&quot;   &quot;ACTB&quot;   &quot;CCT4&quot;   &quot;RDX&quot;   
##  [9] &quot;GNA13&quot;  &quot;CTNNB1&quot;
## 
## $SR_exosome
## [1] &quot;HSP70&quot; &quot;HSP90&quot; &quot;CD9&quot;   &quot;CD63&quot;  &quot;CD81&quot;  &quot;CD82&quot; 
## 
## $Positive_regulation_of_exosomal_secretion
##  [1] &quot;ATP13A2&quot; &quot;CHMP2A&quot;  &quot;HGS&quot;     &quot;MYO5B&quot;   &quot;PDCD6IP&quot; &quot;RAB7&quot;    &quot;SDC1&quot;   
##  [8] &quot;SDC4&quot;    &quot;SDCBP&quot;   &quot;SMPD3&quot;   &quot;SNF8&quot;    &quot;STAM&quot;    &quot;TSG101&quot;  &quot;VPS4A&quot;  
## 
## $Negative_regulation_of_exosomal_secretion
## [1] &quot;VPS4B&quot; &quot;PRKN&quot;  &quot;RAB7&quot; 
## 
## $Exosomal_secretion
## [1] &quot;STEAP3&quot; &quot;TSG101&quot; &quot;RAB11A&quot; &quot;RAB27A&quot; &quot;COPS5&quot; 
## 
## $Exosome_assembly
## [1] &quot;CD34&quot;    &quot;PDCD6IP&quot; &quot;SDC1&quot;    &quot;SDC4&quot;    &quot;SDCBP&quot;   &quot;STAM&quot;    &quot;TSG101&quot; 
## 
## $Extracellular_vesicle_biogenesis
##  [1] &quot;ARRDC1&quot;  &quot;ARRDC4&quot;  &quot;ATP13A2&quot; &quot;CD34&quot;    &quot;CHMP2A&quot;  &quot;COPS5&quot;   &quot;HGS&quot;    
##  [8] &quot;MYO5B&quot;   &quot;PDCD6IP&quot; &quot;PRKN&quot;    &quot;RAB7&quot;    &quot;RAB11A&quot;  &quot;RAB27A&quot;  &quot;SDC1&quot;   
## [15] &quot;SDC4&quot;    &quot;SDCBP&quot;   &quot;SMPD3&quot;   &quot;SNF8&quot;    &quot;STAM&quot;    &quot;STEAP3&quot;  &quot;TSG101&quot; 
## [22] &quot;VPS4B&quot;  
## 
## $MC_Review_Exosome1
##  [1] &quot;TSG101&quot;  &quot;CD9&quot;     &quot;CD81&quot;    &quot;CD63&quot;    &quot;FLOT1&quot;   &quot;ITGB1&quot;   &quot;ITGA1&quot;  
##  [8] &quot;HSP70&quot;   &quot;AIP1&quot;    &quot;ALIX&quot;    &quot;PDCD6IP&quot;
## 
## $MC_Review_Exosome2
##  [1] &quot;RAB27A&quot;  &quot;RAB27B&quot;  &quot;PIKFYVE&quot; &quot;HRS&quot;     &quot;SYT7&quot;    &quot;CTTN&quot;    &quot;STAT3&quot;  
##  [8] &quot;PKM2&quot;    &quot;UNC13D&quot;  &quot;miR-155&quot; &quot;EGFR&quot;    &quot;RAS&quot;     &quot;EIF3C&quot;   &quot;LKB1&quot;   
## [15] &quot;STK11&quot;  
## 
## $CMLS_Review_Exosome
##  [1] &quot;HRS&quot;      &quot;STAM1&quot;    &quot;TSG101&quot;   &quot;CHMP4C&quot;   &quot;ALIX&quot;     &quot;VAT1&quot;    
##  [7] &quot;VPS4&quot;     &quot;CD9&quot;      &quot;CD82&quot;     &quot;CD63&quot;     &quot;LMP1&quot;     &quot;TSPAN8&quot;  
## [13] &quot;VAMP7&quot;    &quot;YKT6&quot;     &quot;PKM2&quot;     &quot;SNAP-23&quot;  &quot;RALA&quot;     &quot;RALB&quot;    
## [19] &quot;RAB2B&quot;    &quot;RAB5A&quot;    &quot;RAB9A&quot;    &quot;RAB7&quot;     &quot;RAB11&quot;    &quot;RAB27A&quot;  
## [25] &quot;RAB27B&quot;   &quot;RAB35&quot;    &quot;DGKA&quot;     &quot;PLD2&quot;     &quot;ARF6&quot;     &quot;ATG12&quot;   
## [31] &quot;ATG7&quot;     &quot;PIKFYVE&quot;  &quot;BST2&quot;     &quot;ATP6V0A4&quot;
## 
## $Ferroptosis
##  [1] &quot;ACSL4&quot;      &quot;AKR1C1-3&quot;   &quot;ALOXs&quot;      &quot;ATP5G3&quot;     &quot;CARS&quot;      
##  [6] &quot;CBS&quot;        &quot;CD44v&quot;      &quot;CHAC1&quot;      &quot;CISD1&quot;      &quot;CS&quot;        
## [11] &quot;DPP4&quot;       &quot;FANCD2&quot;     &quot;GCLC/GCLM&quot;  &quot;GLS2&quot;       &quot;GPX4&quot;      
## [16] &quot;GSS&quot;        &quot;HMGCR&quot;      &quot;HSPB1/5&quot;    &quot;KOD&quot;        &quot;LPCAT3&quot;    
## [21] &quot;MT1G&quot;       &quot;NCOA4&quot;      &quot;NFE2L2&quot;     &quot;PTGS2&quot;      &quot;RPL8&quot;      
## [26] &quot;SAT1&quot;       &quot;SLC7A11&quot;    &quot;SQS&quot;        &quot;TFRC&quot;       &quot;TP53&quot;      
## [31] &quot;TTC35/EMC2&quot; &quot;MESH1&quot;     
## 
## $EV_Cell_2020
##  [1] &quot;HSP90AB1&quot; &quot;HSP90AA1&quot; &quot;CD9&quot;      &quot;ALIX&quot;     &quot;FLOT1&quot;    &quot;FLOT2&quot;   
##  [7] &quot;TSG101&quot;   &quot;HSPA8&quot;    &quot;CD81&quot;     &quot;CD63&quot;     &quot;HBB&quot;      &quot;JCHAIN&quot;  
## [13] &quot;A2M&quot;      &quot;B2M&quot;      &quot;FN1&quot;      &quot;RAP1B&quot;    &quot;LGALS3BP&quot; &quot;GSN&quot;     
## [19] &quot;MSN&quot;      &quot;FLNA&quot;     &quot;ACTB&quot;     &quot;STOM&quot;     &quot;PRDX2&quot;   
## 
## $CD8
## [1] &quot;CD8A&quot;   &quot;CXCL10&quot; &quot;CXCL9&quot;  &quot;GZMA&quot;   &quot;GZMB&quot;   &quot;IFNG&quot;   &quot;PRF1&quot;   &quot;TBX21&quot;</code></pre>
</div>
</div>
<div id="how-to-export-gene-signature" class="section level2 hasAnchor" number="4.9">
<h2><span class="header-section-number">4.9</span> How to export gene signature<a href="signature-score-calculation.html#how-to-export-gene-signature" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Using the <code>output_sig</code> function, user can export the signatures of the list structure to a csv file for other purposes. This step is exactly the reverse of <code>format_signatures</code>.</p>
<div class="sourceCode" id="cb85"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb85-1"><a href="signature-score-calculation.html#cb85-1" tabindex="-1"></a>sig <span class="ot">&lt;-</span> <span class="fu">output_sig</span>(<span class="at">signatures =</span> signature_sc, <span class="at">format =</span> <span class="st">&quot;csv&quot;</span>, <span class="at">file.name =</span> <span class="st">&quot;sc_signature&quot;</span>)</span>
<span id="cb85-2"><a href="signature-score-calculation.html#cb85-2" tabindex="-1"></a>sig[<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span></code></pre></div>
<pre><code>##   CD4_c0_Tcm CD4_c1_Treg CD4_c10_Tn_LEF1_ANKRD55 CD4_c11_Tisg CD4_c2_Tn
## 1      ANXA1       FOXP3                 ANKRD55        ISG15    NBEAL1
## 2       LMNA       IL2RA                    LEF1         IFI6      CCR7
## 3        VIM     TNFRSF4                    TCF7       IFI44L   GLTSCR2
## 4      KLRB1       TIGIT                   NOSIP          MX1      TCF7
## 5       IL7R      CARD16                    SELL        IFIT3    GNB2L1
## 6      ZFP36    TNFRSF18                   IL6ST        IFIT1      SELL
## 7    ZFP36L2        BATF                 LDLRAP1        RSAD2   C6orf48
## 8     GPR183       CTLA4                  RIPOR2        STAT1    TMEM66</code></pre>
</div>
<div id="references-1" class="section level2 hasAnchor" number="4.10">
<h2><span class="header-section-number">4.10</span> References<a href="signature-score-calculation.html#references-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>ssgsea</strong>: Barbie, D.A. et al (2009). Systematic RNA interference reveals that oncogenic KRAS-driven cancers require TBK1. Nature, 462(5):108-112.</p>
<p><strong>gsva</strong>: H√§nzelmann, S., Castelo, R. and Guinney, J. (2013). GSVA: Gene set variation analysis for microarray and RNA-Seq data. BMC Bioinformatics, 14(1):7.</p>
<p><strong>zscore</strong>: Lee, E. et al (2008). Inferring pathway activity toward precise disease classification. PLoS Comp Biol, 4(11):e1000217.</p>
<p><strong>PCA method</strong>: Mariathasan S, Turley SJ, Nickles D, et al.¬†TGFŒ≤ attenuates tumour response to PD-L1 blockade by contributing to exclusion of T cells. Nature. 2018 Feb 22;554(7693):544-548.</p>
<p><strong>MSigDB</strong>:Dolgalev I (2022). msigdbr: MSigDB Gene Sets for Multiple Organisms in a Tidy Data Format. R package version 7.5.1. (<a href="https://www.gsea-msigdb.org/gsea/msigdb/" class="uri">https://www.gsea-msigdb.org/gsea/msigdb/</a>)</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="rna-data-preprocessing.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="tme-deconvolution.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/DongqiangZeng0808/signature-analysis.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
